### DISCUSSION TITLE
Error in running a phase field file

### DISCUSSION THREAD

**Discussion by mdh23rct** [idaholab/moose#25189] [Q&A Modules: Phase field]
Dear all,
Hope you are doing great! I am a new user of MOOSE. Currently, I am coding to simulate phase field. But, while want to run the file, it is showing the following error:
```
A 'SwitchingFunctionMaterial' is not a registered object.

If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable. 
```
Would you be kind to let me know what is the problem, and how I can solve the problem!!
I appreciate your help.

Best regards,
Zahid

**Comment by GiudGiud** [comment]
Hello

Which executable are you using?
You need to be using the phase field module executable, or the executable from an application that includes this module

Guillaume

**Reply by mdh23rct** [reply]
Dear Dr. Guillaume,
I appreciate your reply. What I am seeing, in the application folder (\\wsl.localhost\Ubuntu\home\user\projects\irradiation), there are no modules, though the modules are in the moose folder (\\wsl.localhost\Ubuntu\home\user\projects\moose). Unfortunately, the Makefile in the application folder cannot link to the modules of the moose folder. What to do then? 
 
The make file in the application folder look like this:
```
##############################################################################
################### MOOSE Application Standard Makefile #######################
###############################################################################
#
# Optional Environment variables
# MOOSE_DIR        - Root directory of the MOOSE project
#
###############################################################################
# Use the MOOSE submodule if it exists and MOOSE_DIR is not set
MOOSE_SUBMODULE    := $(CURDIR)/moose
ifneq ($(wildcard $(MOOSE_SUBMODULE)/framework/Makefile),)
  MOOSE_DIR        ?= $(MOOSE_SUBMODULE)
else
  MOOSE_DIR        ?= $(shell dirname `pwd`)/moose
endif

# framework
FRAMEWORK_DIR      := $(MOOSE_DIR)/framework
include $(FRAMEWORK_DIR)/build.mk
include $(FRAMEWORK_DIR)/moose.mk

################################## MODULES ####################################
# To use certain physics included with MOOSE, set variables below to
# yes as needed.  Or set ALL_MODULES to yes to turn on everything (overrides
# other set variables).

ALL_MODULES                 := yes

CHEMICAL_REACTIONS          := no
CONTACT                     := no
ELECTROMAGNETICS            := no
EXTERNAL_PETSC_SOLVER       := no
FLUID_PROPERTIES            := no
FSI                         := no
FUNCTIONAL_EXPANSION_TOOLS  := no
GEOCHEMISTRY                := no
HEAT_CONDUCTION             := no
LEVEL_SET                   := no
MISC                        := no
NAVIER_STOKES               := no
OPTIMIZATION                := no
PERIDYNAMICS                := no
PHASE_FIELD                 := no
POROUS_FLOW                 := no
RAY_TRACING                 := no
REACTOR                     := no
RDG                         := no
RICHARDS                    := no
STOCHASTIC_TOOLS            := no
THERMAL_HYDRAULICS          := no
TENSOR_MECHANICS            := no
XFEM                        := no

include $(MOOSE_DIR)/modules/modules.mk
###############################################################################

# dep apps
APPLICATION_DIR    := $(CURDIR)
APPLICATION_NAME   := irradiation
BUILD_EXEC         := yes
GEN_REVISION       := no
include            $(FRAMEWORK_DIR)/app.mk

###############################################################################
# Additional special case targets should be added here
```
I am using the command: ./irradiation-opt -i irradiation.i   to run the file. So irradiation-opt should be the executable. Am I right!

Thanks,
Zahid

**Reply by richmondodufisan** [reply]
I think you need to change PHASE FIELD from no to yes in the Makefile in your application, and then recompile. It should be able to find it after that.

**Reply by GiudGiud** [reply]
Exactly that. You may need to `make clobberall` before re-make-ing.

**Reply by mdh23rct** [reply]
Thank you guys for helping me out. Last time, I have given ALL_MODULES := yes. It should turn on all the modules as instructed in the MOOSE help. In addition, this time also made PHASE_FIELD := yes. What I found the same file runs on the first try, but may not run on the second try. I am using WSL shell command. I am not exactly sure what is happening. I am thinking of bugs in the compiler!!

Do you have any comments on that!
